apps/resnet_50/Resnet50Generator.cpp

```diff
@@ -0,0 +1,383 @@
+            if (br1_i >= 0) {
+                br1_conv[br1_i] = conv2D(br2a_input, br1_ws[br1_i], br1_conv_weights[br1_i], "br1_conv");
+                br1_norm[br1_i] = norm_layer(br1_conv[br1_i], br1_mu[br1_i], br1_sig[br1_i], "br1_norm");
+                br1_scale[br1_i] = scale_layer(br1_norm[br1_i], br1_gamma[br1_i], br1_beta[br1_i], "br1_scale");
+                resunit_sum_input = br1_scale[br1_i];
+            } else {
+                resunit_sum_input = resunit_relu[block_id-1];
+            }
+
+            // branch2a
+            auto weights = br2a_conv_weights[block_id];
+
+            br2a_conv[block_id] = conv2D(br2a_input, br2a_ws[block_id], weights, "block" + std::to_string(block_id) + "_2a_conv");
+            br2a_norm[block_id] = norm_layer(br2a_conv[block_id], br2a_mu[block_id], br2a_sig[block_id], "block" + std::to_string(block_id) + "_2a_norm");
+            br2a_scaled[block_id] = scale_layer(br2a_norm[block_id], br2a_gamma[block_id], br2a_beta[block_id], "block" + std::to_string(block_id) + "_2a_scale");
+            br2a_relu[block_id] = relu_layer(br2a_scaled[block_id], "2a_relu");
+
+            // branch 2b
+            weights = br2b_conv_weights[block_id];
+            br2b_conv[block_id] = conv2D(br2a_relu[block_id], br2b_ws[block_id], weights, "block" + std::to_string(block_id) + "_2b_conv");
+            br2b_norm[block_id] = norm_layer(br2b_conv[block_id], br2b_mu[block_id], br2b_sig[block_id], "block" + std::to_string(block_id) + "_2b_norm");
+            br2b_scaled[block_id] = scale_layer(br2b_norm[block_id], br2b_gamma[block_id], br2b_beta[block_id], "block" + std::to_string(block_id) + "_2b_scale");
+            br2b_relu[block_id] = relu_layer(br2b_scaled[block_id], "2b_relu");
+
+            // branch 2c
+            weights = br2c_conv_weights[block_id];
+            br2c_conv[block_id] = conv2D(br2b_relu[block_id], br2c_ws[block_id], weights, "block" + std::to_string(block_id) + "_2c_conv");
+            br2c_norm[block_id] = norm_layer(br2c_conv[block_id], br2c_mu[block_id], br2c_sig[block_id], "block" + std::to_string(block_id) + "_2c_norm");
+            br2c_scaled[block_id] = scale_layer(br2c_norm[block_id], br2c_gamma[block_id], br2c_beta[block_id], "block" + std::to_string(block_id) + "_2c_scale");
+
+            // create residual unit
+            resunit_sum[block_id] = sum_layer(resunit_sum_input, br2c_scaled[block_id], "block" + std::to_string(block_id) + "_res_sum");
+            resunit_relu[block_id] = relu_layer(resunit_sum[block_id], "block" + std::to_string(block_id) + "_res_relu");
+
+            // create final 3 layers
+            if (block_id == 15) {
+                pool5 = avg_pool_layer(resunit_relu[block_id], pool5_ws, "pool5");
+                fc1000 = fc_layer(pool5, fc1000_ws, fc1000_weights, fc1000_bias, "fc");
+                final_output = softmax_layer(fc1000, 1000, "softmax");
+            }
+        }
+        conv1.f.compute_root();
```

Comment:
There isn't really a schedule here yet, just enough of one to compile. I'm worried people will advertise that they are X times faster than Halide without realizing this. Please add:
```
// TODO: Actually schedule this. 
```

and where it prints the runtime in process.cpp please make it also print "This code hasn't been scheduled properly yet so this runtime isn't representative of anything and should not be used as a basis for any comparisons".

When I get around to actually writing a schedule I'll remove the prints.

Reply:
Here you go...
```
$ make benchmark_and_validate
bin/process 10  ./bin/pytorch_weights/ 123 ./bin/res50gen_output.bin
Running Resnet50 for 10 iterations....
*************************** Please note ******************************
This code hasn't been scheduled properly yet so this runtime
isn't representative of anything and should not be used as a basis
for any comparisons.
Execution time : 357.742ms
**********************************************************************
Class for random data of seed 123 is 783
Writing output layer to ./bin/res50gen_output.bin
python3 validate_resnet50_output.py ./bin/res50gen_output.bin 123
Success!
```